### PR TITLE
[ja] update testing.rst

### PR DESCRIPTION
**Pull request by okinaka**
Refs #4091

### CODE REVIEW

**Comment by jeremyharris** [comment]
Whoops, thanks!

File: ja/development/testing.rst [hunk 1]
```diff
@@ -1340,6 +1340,106 @@ PagematronComponent というコンポーネントがアプリケーションに
...
 
+.. _testing-events:
+
+イベントのテスト
+================
+
+:doc:`/core-libraries/events` は、アプリケーションコードを分離する素晴らしい方法ですが、
+テストの際、これらのイベントを実行するテストケース内のイベントの結果をテストすることになります。
+これは、 ``assertEventFired`` や ``assertEventFiredWith`` を使用することで削除可能な
+結合の代わりになる付加的な形です。
```
Comment (chinpei215) [inline ja/development/testing.rst]: >  This is an additional form of coupling that can be removed by using ``assertEventFired`` and ``assertEventFiredWith``
instead.

In this sentence, `coupling` would be [a technical term](https://en.wikipedia.org/wiki/Coupling_(computer_programming)). An adjective `additional` modifiers a noun phrase `form of coupling`, `that ~` is an adjective clause that modifiers `an additional form of coupling`, and an adverb `instead` modifiers `using ~`.

（この文では `coupling` は専門用語の結合のことでしょう。形容詞 `additional` は名詞句 `form of coupling` を修飾する形容詞で、`that ~` 以下は `an additional form of coupling` を修飾する形容詞節、そして副詞 `instead` は `using ~` 以下を修飾しています）

```
これは、``assertEventFired`` や ``assertEventFiredWith`` を代わりに使うことで削除ができる、余分な結合の一種です。
```

File: ja/development/testing.rst [hunk 2]
```diff
@@ -1340,6 +1340,106 @@ PagematronComponent というコンポーネントがアプリケーションに
 他のヘルパーを使用するヘルパーをテストしている時、View クラスの ``loadHelpers`` メソッドを
 モックにしてください。
 
+.. _testing-events:
+
+イベントのテスト
+================
+
+:doc:`/core-libraries/events` は、アプリケーションコードを分離する素晴らしい方法ですが、
+テストの際、これらのイベントを実行するテストケース内のイベントの結果をテストすることになります。
```
Comment (chinpei215) [inline ja/development/testing.rst]: > you tend to test
 
テストすることになりがちです。

File: ja/development/testing.rst [hunk 3]
```diff
@@ -1340,6 +1340,106 @@ PagematronComponent というコンポーネントがアプリケーションに
...
+        }
+    }
+
+.. note::
+    イベントの発生をアサートするために、イベントマネージャ上で最初に :ref:`tracking-events`
+    を有効にしなければなりません。
+
+上記の ``OrdersTable`` をテストするために、イベントの発生をアサートする際に ``setUp()``
+内でトラッキングを有効にします。そして、 ``$order`` エンティティをイベントデータに
+渡してアサートしてください。 ::
```
Comment (chinpei215) [inline ja/development/testing.rst]: > To test the ``OrdersTable`` above, we enable tracking in ``setUp()`` then assert that the event was fired, and assert that the ``$order`` entity was passed in the event data

`assert` is a verb. So the nominative is probably `we`.

(`asset` は動詞です。なので、主語はたぶん `we` です)
```
上記の ``OrdersTable`` をテストするために、``setUp()`` 内でトラッキングを有効にした後、
イベントが発生することをアサートし、そして ``$order`` エンティティがイベントデータに渡されることをアサートします。
```

we enable trakcing in `setUp()`:
```php
$this->Orders->getEventManager()->setEventList(new EventList());
```

(we) assert that the event was fired:
```php
$this->assertEventFired('Model.Order.afterPlace', $this->Orders->getEventManager());
```

(we) assert that the ``$order`` entity was passed in the event data:
```php
$this->assertEventFiredWith('Model.Order.afterPlace', 'order', $order, $this->Orders->getEventManager());
```

**Comment by okinaka** [comment]
@chinpei215 Thanks! I updated.

**Comment by chinpei215** [comment]
@okinaka Nice work!

**Pull request merged**